packages/discussions/graphql/resolvers/Comment.js

```diff
@@ -242,6 +242,9 @@ module.exports = {
     } else { // FORBIDDEN or ALLOWED
       if (commenterPreferences && commenterPreferences.anonymous != null) {
         anonymous = commenterPreferences.anonymous
+        if (anonymous && commenterPreferences.anonymousDifferentiator !== null) {
+          anonymousComment.name += ` ${commenterPreferences.anonymousDifferentiator}`
+        }
```

Comment:
If `anonymousDifferentiator` is set in `submitComment` we could move this after the whole if else block and also set it for `ENFORCED` discussions.